test/stdlib/CaseIterableTests.swift

```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-simple-swift %t
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var CaseIterableTests = TestSuite("CaseIterableTests")
+
+CaseIterableTests.test("Simple Enums") {
+  enum SimpleEnum: CaseIterable {
+    case bar
+    case baz
+    case quux
+  }
+
+  expectEqual(SimpleEnum.allCases.count, 3)
```

Comment:
What's `type(of: SimpleEnum.allCases)` here? Is it just `Array<SimpleEnum>`?

Reply:
For now. I hope we can come up with something more creative.